webapp/portlet/src/main/webapp/activity-composer-app/components/ExoActivityComposer.vue

```diff
@@ -82,6 +123,15 @@ export default {
   },
   created() {
     this.activityComposerActions = getActivityComposerExtensions();
+    this.activityComposerActions.forEach(action => {
+      if (action.component) {
+        this.actionsData[action.key] = action.component.model;
+        this.actionsEvents[action.key] = action.component.events;
+      }
+    });
+    this.$watch('actionsData.file.value', function () {
```

Comment:
`file` is still "hardcoded" here, can we get rid of it ?

Reply:
So what we will put? what we will be watching to put in the attachments array?

Reply:
We could add a new event in the ECMS plugin to call a method addAttachment (to be defined in the ExoActivityComposer component):
```
        {
          'event': 'uploadingFileFinished',
          'listener': 'addAttachment',
          'listenerParam': 'attachment'
        }
```
so we can remove this watch. WDYT ?

Reply:
I suggest making a watch to the attachments array in the attachments component because we need the exact number to display and a loader to be displayed. Otherwise, we will be creating three events(finishUpload, remove, addExisting) for handling the attachments from the new composer.